### PR TITLE
Implement native fuzzing in Go

### PR DESCRIPTION
**Pull request by hanzei**
Fixes https://github.com/agnivade/levenshtein/issues/27

### CODE REVIEW

**Comment by agnivade** [comment]
Oh didn't expect you to be here! Is this ready for review?

**Comment by hanzei** [comment]
Yeah, this is my first look into Fuzzing in Go. Please let me know what you think.

File: levenshtein_test.go [hunk 1]
```diff
@@ -120,3 +120,45 @@ func BenchmarkAll(b *testing.B) {
...
+		f.Add(tc.a, tc.b)
+	}
+	f.Fuzz(func(t *testing.T, a, b string) {
+		n := agnivade.ComputeDistance(a, b)
+		if n < 0 {
+			t.Errorf("Distance can not be negative: %d, a: %q, b: %q", n, a, b)
+		}
+		if n > len(a)+len(b) {
+			t.Errorf("Distance can not be greater than sum of lengths of a and b: %d, a: %q, b: %q", n, a, b)
+		}
```
Comment (hanzei) [inline levenshtein_test.go:144]: I was hoping to check more properties, but I could only come up with this list. Is there more to check?

**Comment by agnivade** [comment]
Thanks! I will look at this some time.

**Comment by hanzei** [comment]
I'm not in a hurry :wink:

File: levenshtein_test.go (hunk 1 above)
Comment (agnivade) [reply levenshtein_test.go:166]: This looks good Ben!

File: levenshtein_test.go [hunk 2]
```diff
@@ -142,3 +142,45 @@ func BenchmarkAll(b *testing.B) {
...
+
+// Fuzzing
+// ----------------------------------------------
+
+func FuzzComputeDistanceDifferent(f *testing.F) {
+	testcases := []struct{ a, b string }{
+		{"levenshtein", "frankenstein"},
+		{"resumé and café", "resumés and cafés"},
+		{"Hafþór Júlíus Björnsson", "Hafþor Julius Bjornsson"},
+		{"།་གམ་འས་པ་་མ།", "།་གམའས་པ་་མ"},
```
Comment (agnivade) [inline levenshtein_test.go:154]: I am wondering if we could add a few more test cases from junk binary strings? That should make the corpus a bit more healthy and let the fuzzer generate smarter inputs.

I just generated a few random byte strings `_p~𕍞`, `b잖PwN`, `7ȪJR`, `6L)wӝ`, `Y>q8օ݌`. Can we add them to the test cases?

**Review by agnivade — COMMENTED** [review]
Later than expected, but not forgotten!

Just a small comment Ben, and thanks again for this!

File: levenshtein_test.go (hunk 2 above)
Comment (hanzei) [reply levenshtein_test.go:154]: Sounds good, added three more cases.

**Review by agnivade — APPROVED** [review]
Thanks!

**Pull request merged**